### PR TITLE
Day 7 (pagination and create) 

### PR DESCRIPTION
**Pull request by vdumbrav**

### CODE REVIEW

File: src/Controller/CategoryController.php [hunk 1]
```diff
@@ -14,22 +16,33 @@
...
-     */
-    public function list(CategoryRepository $categoryRepository): Response
-    {
-        return $this->render('category/index.html.twig', ['categories' => $categoryRepository->findAll()]);
-    }
-
-    /**
-     * @Route("/{slug}", name="category_show", methods="GET")
+     *
+     * @Route("/{slug}/{page}", name="category.show", defaults={"page":1})
```
Comment (gregurco) [inline src/Controller/CategoryController.php]: add requirements for page. Only number.

File: src/Controller/CategoryController.php [hunk 2]
```diff
@@ -14,22 +16,33 @@
...
+     *
      * @return Response
      */
-    public function show(Category $category): Response
+
+    public function show(
+        Category $category,
+        int $page,
+        PaginatorInterface $paginator
+    ): Response
```
Comment (gregurco) [inline src/Controller/CategoryController.php]: sing PSR it should be:
```
): Response {
```

https://github.com/php-fig/fig-standards/blob/master/proposed/extended-coding-style-guide.md#45-method-and-function-arguments

File: src/Controller/CategoryController.php [hunk 3]
```diff
@@ -14,22 +16,33 @@
...
+        $activeJobs = $paginator->paginate(
+            $this->getDoctrine()
+                ->getRepository(Job::class)
+                ->getPaginatedActiveJobsByCategoryQuery($category),
+            $page,
+            $this->getParameter('max_job_on_category')
+        );
+        return $this->render('category/show.html.twig', array(
+            'category' => $category,
+            'activeJobs' => $activeJobs
```
Comment (gregurco) [inline src/Controller/CategoryController.php]: add `,` at the last element in array

File: src/Repository/JobRepository.php [hunk 4]
```diff
@@ -25,4 +27,18 @@ public function findActiveJob()
             ->getQuery()
             ->getResult();
     }
+
+    public function getPaginatedActiveJobsByCategoryQuery(Category $category): AbstractQuery
```
Comment (gregurco) [inline src/Repository/JobRepository.php]: add PhpDoc for this method

File: src/Repository/JobRepository.php [hunk 5]
```diff
@@ -25,4 +27,18 @@ public function findActiveJob()
...
+    {
+        return $this->createQueryBuilder('job')
+            ->where('job.category = :category')
+            ->andWhere('job.expiresAt > :date')
+            ->andWhere('job.activated = :activated')
+            ->setParameters([
+                'category' => $category,
+                'date' => new \DateTime(),
+                'activated' => true
```
Comment (gregurco) [inline src/Repository/JobRepository.php]: also `,` at last element

**Review by gregurco — CHANGES_REQUESTED** [review]

**Comment by vdumbrav** [comment]
@gregurco fixed.

File: src/Controller/CategoryController.php [hunk 6]
```diff
@@ -14,22 +16,32 @@
...
-     */
-    public function list(CategoryRepository $categoryRepository): Response
-    {
-        return $this->render('category/index.html.twig', ['categories' => $categoryRepository->findAll()]);
-    }
-
-    /**
-     * @Route("/{slug}", name="category_show", methods="GET")
+     *
+     * @Route("/{slug}/{page}", name="category.show", defaults={"page":1}, requirements={"id" = "\d+"})
```
Comment (gregurco) [inline src/Controller/CategoryController.php]: `requirements={"page" = "\d+"}` - not id, page

File: src/Controller/JobController.php [hunk 7]
```diff
@@ -29,12 +33,39 @@ public function list(): Response
...
+    /**
+     * @Route("/create", name="job.create")
+     * @Method({"GET", "POST"})
+     *
+     * @param Request $request
+     * @param EntityManagerInterface $em
+     *
+     * @return Response
+     */
+    public function createAction(Request $request, EntityManagerInterface $em): Response
```
Comment (gregurco) [inline src/Controller/JobController.php]: I see that you don't use `Action` in methods.. remove it

File: src/Controller/JobController.php [hunk 8]
```diff
@@ -29,12 +33,39 @@ public function list(): Response
...
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $em->persist($job);
+            $em->flush();
+
+            return $this->redirectToRoute('job.list');
+        }
+
+        return $this->render('job/create.html.twig', [
+            'form' => $form->createView()
```
Comment (gregurco) [inline src/Controller/JobController.php]: add `,` at last element in array

File: src/Form/JobType.php [hunk 9]
```diff
@@ -2,34 +2,117 @@
...
+                    'No' => false,
+                ],
+                'constraints' => [
+                    new NotNull(),
+                ]
+            ])
+            ->add('email', EmailType::class, [
+                'constraints' => [
+                    new NotBlank(),
+                    new Email()
```
Comment (gregurco) [inline src/Form/JobType.php]: also `,`

File: templates/job/create.html.twig [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    <div class="container">
+        <h1>Job creation</h1>
+
+        {{ form_start(form, {'attr': {'novalidate': 'novalidate'}}) }}
+        {{ form_widget(form) }}
+
+        <div class="form-group">
+            <button type="submit" class="btn btn-default">Create</button>
+        </div>
+        {{ form_end(form) }}</div>
```
Comment (gregurco) [inline templates/job/create.html.twig]: template style:
```
{% extends 'base.html.twig' %}

{% block body %}
    <div class="container">
        <h1>Job creation</h1>

        {{ form_start(form, {'attr': {'novalidate': 'novalidate'}}) }}
            {{ form_widget(form) }}

            <div class="form-group">
                <button type="submit" class="btn btn-default">Create</button>
            </div>
        {{ form_end(form) }}
    </div>
{% endblock %}
```

**Review by gregurco — CHANGES_REQUESTED** [review]

File: src/Controller/CategoryController.php [hunk 11]
```diff
@@ -14,22 +16,32 @@
...
+        PaginatorInterface $paginator
+    ): Response {
+        $activeJobs = $paginator->paginate(
+            $this->getDoctrine()
+                ->getRepository(Job::class)
+                ->getPaginatedActiveJobsByCategoryQuery($category),
+            $page,
+            $this->getParameter('max_job_on_category')
+        );
+        return $this->render('category/show.html.twig', array(
```
Comment (gregurco) [inline src/Controller/CategoryController.php]: use `[]` instead of `array()`

**Review by gregurco — CHANGES_REQUESTED** [review]

**Comment by vdumbrav** [comment]
@gregurco fixed

**Pull request merged**